packages/synapse-react-client/src/components/ColumnFilter/ColumnFilter.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import { useQueryContext } from '../QueryContext/QueryContext'
+import {
+  FacetColumnResultValues,
+  FacetColumnValuesRequest,
+} from '@sage-bionetworks/synapse-types'
+import { useQuery } from '@tanstack/react-query'
+import { Autocomplete, TextField, Box } from '@mui/material'
+import { getCorrespondingSelectedFacet } from '../../utils/functions/queryUtils'
+
+export type FilterProps = {
+  filterColumnName?: string
+  addValueToSelectedFacet?: (
+    facet: { columnName: string },
+    value: string,
+  ) => void
+  removeSelectedFacet?: (facet: { columnName: string }) => void
+}
+
+function ColumnFilter(props: FilterProps) {
+  const queryContext = useQueryContext()
+  const {
+    executeQueryRequest,
+    queryMetadataQueryOptions,
+    getCurrentQueryRequest,
+    addValueToSelectedFacet,
+    removeSelectedFacet,
+  } = queryContext
+  const { data: queryMetadata } = useQuery(queryMetadataQueryOptions)
+  const { filterColumnName } = props
+
+  // const addValueToSelectedFacet = queryContext.addValueToSelectedFacet
+  // const addValueToSelectedFacet = queryContext.removeSelectedFacet
+
+  const currentQuery = getCurrentQueryRequest()
+
+  const activeFacet = (queryMetadata?.facets ?? []).find(
+    facet => facet.columnName === `${filterColumnName}`,
+  ) as FacetColumnResultValues | undefined
+
+  // Map the facet values to format for MUI Autocomplete options
+  const filterOptions = activeFacet?.facetValues.map(({ value }) => value) ?? []
+
+  const selectedFacetFromQuery = getCorrespondingSelectedFacet(
+    { columnName: filterColumnName || '' },
```

Comment:
`filterColumnName` is always defined (after you update the prop type)
```suggestion
    { columnName: filterColumnName },
```